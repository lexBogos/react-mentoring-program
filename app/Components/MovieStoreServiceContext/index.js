import {
    MovieStoreServiceProvider,
    MovieStoreServiceConsumer
  } from './movie-store-service-context';
  
  export {
    MovieStoreServiceProvider,
    MovieStoreServiceConsumer
  };
  