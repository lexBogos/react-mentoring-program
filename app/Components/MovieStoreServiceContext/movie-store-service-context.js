import React from 'react';

const {
  Provider: MovieStoreServiceProvider,
  Consumer: MovieStoreServiceConsumer
} = React.createContext();

export {
  MovieStoreServiceProvider,
  MovieStoreServiceConsumer
};
