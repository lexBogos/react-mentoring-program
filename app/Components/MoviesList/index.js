import React from 'react';
import MovieCard from '../MovieCard';
import './index.scss';

const movies = [
    {
        title: 'Goodfellas',
        description: 'The story of Henry Hill and his life in the mob, covering his relationship with his wife Karen Hill and his mob partners Jimmy Conway and Tommy DeVito in the Italian-American crime syndicate.',
        poster: 'https://m.media-amazon.com/images/M/MV5BY2NkZjEzMDgtN2RjYy00YzM1LWI4ZmQtMjIwYjFjNmI3ZGEwXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_.jpg',
        id: 'tt0099685',
        year: 1990,
        type: 'Biography, Crime, Drama',
    },
    {
        title: 'The Matrix',
        description: 'When a beautiful stranger leads computer hacker Neo to a forbidding underworld, he discovers the shocking truth--the life he knows is the elaborate deception of an evil cyber-intelligence.',
        poster: 'https://m.media-amazon.com/images/M/MV5BNzQzOTk3OTAtNDQ0Zi00ZTVkLWI0MTEtMDllZjNkYzNjNTc4L2ltYWdlXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_FMjpg_UY681_.jpg',
        id: 'tt0133093',
        year: 1999,
        type: 'Action, Sci-Fi',
    },
    {
        title: 'Terminator 2: Judgment Day',
        description: 'A cyborg, identical to the one who failed to kill Sarah Connor, must now protect her ten year old son, John Connor, from a more advanced and powerful cyborg.',
        poster: 'https://m.media-amazon.com/images/M/MV5BMGU2NzRmZjUtOGUxYS00ZjdjLWEwZWItY2NlM2JhNjkxNTFmXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_FMjpg_UY403_.jpg',
        id: 'tt0103064',
        year: 1991,
        type: 'Action, Sci-Fi',
    },
    {
        title: 'Modern Times',
        description: 'The Tramp struggles to live in modern industrial society with the help of a young homeless woman.',
        poster: 'https://m.media-amazon.com/images/M/MV5BYjJiZjMzYzktNjU0NS00OTkxLWEwYzItYzdhYWJjN2QzMTRlL2ltYWdlL2ltYWdlXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_FMjpg_UX578_.jpg',
        id: 'tt0027977',
        year: 1936,
        type: 'Comedy, Drama, Family',
    },
    {
        title: 'Memento',
        description: `A man with short-term memory loss attempts to track down his wife's murderer.`,
        poster: 'https://m.media-amazon.com/images/M/MV5BZTcyNjk1MjgtOWI3Mi00YzQwLWI5MTktMzY4ZmI2NDAyNzYzXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_FMjpg_UY218_.jpg 148w, https://m.media-amazon.com/images/M/MV5BZTcyNjk1MjgtOWI3Mi00YzQwLWI5MTktMzY4ZmI2NDAyNzYzXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_FMjpg_UY327_.jpg 222w, https://m.media-amazon.com/images/M/MV5BZTcyNjk1MjgtOWI3Mi00YzQwLWI5MTktMzY4ZmI2NDAyNzYzXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_FMjpg_UY408_.jpg 277w, https://m.media-amazon.com/images/M/MV5BZTcyNjk1MjgtOWI3Mi00YzQwLWI5MTktMzY4ZmI2NDAyNzYzXkEyXkFqcGdeQXVyNjU0OTQ0OTY@._V1_FMjpg_UX1022_.jpg 1022w',
        id: 'tt0209144',
        year: 2000,
        type: 'Mystery, Thriller',
    },
    {
        title: 'Dune',
        description: `Feature adaptation of Frank Herbert's science fiction novel, about the son of a noble family entrusted with the protection of the most valuable asset and most vital element in the galaxy.`,
        poster: 'https://m.media-amazon.com/images/M/MV5BMGFkZGY0NTgtMGEyZC00YzhjLTkyOWItYzMzOTljMDA3ZjU2XkEyXkFqcGdeQXVyNzc4NTU3Njg@._V1_.jpg',
        id: 'tt1160419',
        year: 2021,
        type: 'Adventure, Drama, Sci-Fi',
    },  
]


const MoviesList = () =>
    <div className='movieList'>
        {movies.map(movie => (<MovieCard  
                 title={movie.title}
                 description={movie.description} 
                 key={movie.id}
                 poster={movie.poster}
                 year={movie.year}
                 type={movie.type}
        />))};
    </div>


export default MoviesList;