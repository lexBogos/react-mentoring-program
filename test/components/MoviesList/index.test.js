import React from 'react';
import MoviesList from '../../../app/Components/MoviesList/index';
import '@testing-library/jest-dom/extend-expect';
import { createStore } from "redux";
import reducer from "../../../app/reducers/index";
import { Provider } from "react-redux";
import MovieStoreService from '../../../app/services/movie-store-service';
import { MovieStoreServiceProvider } from '../../../app/components/MovieStoreServiceContext';
import { render as rtlRender } from '@testing-library/react';

function render(
    ui,
    {
      initialState,
      store = createStore(reducer, initialState),
      ...renderOptions
    } = {}
  ) {
    function Wrapper({ children }) {
      return <Provider store={store}>{children}</Provider>
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
  }
  


describe('MoviesList component', () => {

    let onMovieChosen

    const movies = [
        {
            title: 'Goodfellas',
            id: 'tt0099685',
            realiseDate: 1990,
            rate: 4.9
        },
        {
            title: 'Goodfellas',
            id: 'tt0133093',
            realiseDate: 1999,
            rate: 5
        },
        {
            title: 'Terminator 2: Judgment Day',
            id: 'tt0103064',
            realiseDate: 1991,
            rate: 5
        },
        {
            title: 'Modern Times',
            id: 'tt0027977',
            realiseDate: 1936,
            rate: 4.2
        },
        {
            title: 'Memento',
            id: 'tt0209144',
            realiseDate: 2000,
            rate: 4
        },
        {
            title: 'Dune',
            id: 'tt1160419',
            realiseDate: 2021,
            rate: 3.8
        },  
      ];

    beforeEach(()=>{
        onMovieChosen = jest.fn();
    });

    const movieStoreService = new MovieStoreService();

        it('should return component with MoviesList component if modalActionType is "REMOVE_MOVIE"', () => {
            const { asFragment } = render(<MovieStoreServiceProvider value={movieStoreService}><MoviesList/></MovieStoreServiceProvider>, { initialState: { loading:false, chosenMovieIndex: 0, movies: movies, movieStoreService:movieStoreService } });

            expect(asFragment(<MoviesList/>)).toMatchSnapshot();
        });
})