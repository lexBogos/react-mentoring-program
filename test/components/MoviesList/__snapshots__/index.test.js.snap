// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList component should return component with MoviesList component if modalActionType is "REMOVE_MOVIE" 1`] = `
<DocumentFragment>
  <h2>
    Loading...
  </h2>
</DocumentFragment>
`;
